src/main/webapp/resources/js/apis/projects/reference-files.js

```diff
@@ -41,3 +41,23 @@ export async function removeProjectReferenceFile(projectId, fileId) {
       throw new Error(error.response.data.error);
     });
 }
+
+/**
+ * Remove the reference file from the project id provided
+ * @param {projectId} project for which to remove reference file from
+ * @param {formData} List of files to upload
+ * @param config Configuration options for upload
+ * @return {Promise<*>} `data` contains the OK response; `error` contains error information if an error occurred.
+ */
+export async function uploadProjectReferenceFiles(projectId, formData, config) {
+  return await axios
+    .post(
+      setBaseUrl(`ajax/reference-files/project/${projectId}`),
+      formData,
+      config
+    )
+    .then(({ data }) => data)
+    .catch((error) => {
+      throw new Error(error.response.data.error);
+    });
```

Comment:
When using `await` keep to the syntax you like (this is probably how I have been doing it and it is not exactly correct.  Keep the way I have seen you write it, it is much more elegant.  Also, I learned recently that instead of throwing a new error, just return a rejected promise:

```javascript
export async function uploadProjectReferenceFiles(projectId, formData, config) {
  try {
    const { data } = await axios.post(
      setBaseUrl(`ajax/reference-files/project/${projectId}`),
      formData,
      config
    );
    return data;
  } catch (e) {
    return Promise.reject(error.response.data.error);
  }
}
```
You can then handle this the same way:
```javascript
uploadProjectReferenceFiles(projectId, formData, config)
  .then(console.log)
  .catch(console.error);   
```

**You should start freaking out on me when I don't do it this way!**

Reply:
Thanks. Yeah I've found myself flip flopping on it too. Oh for sure!

Reply:
Updated in 496e77453248495380b0e3afe0eea9b5031b7f33